libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -172,15 +172,24 @@ public override async Task<RecognizerResult> RecognizeAsync(DialogContext dc, Sc
                     ((List<Result>)recognizerResult.Properties[ResultProperty]).Insert(0, new Result() { Score = 1.0, Label = new Label() { Name = NoneIntent, Type = LabelType.Intent } });
 
                     // add all scores
-                    foreach (var result in results)
+                    bool duplicateNoneRemoved = false;
+                    for (int i = 0; i < results.Count; ++i)
                     {
-                        if (results[0].Label.Name != NoneIntent || result.Label.Name != NoneIntent)
+                        var result = results[i];
+                        string label = result.Label.Name;
+                        if (!duplicateNoneRemoved && label == NoneIntent && result.Score < 1.0)
+                        {
+                            // remove duplicate None intents
+                            results.RemoveAt(i--);
+                            duplicateNoneRemoved = true;
```

Comment:
What if there are more than 1 None results added into results, besides the one just added due to low top score?

Reply:
From Orchestrator, can we have multiple results with the same Intent label?

Reply:
You are right, Label Resolver does not produce multiple results with the same intent label. However, I still wonder how could the LableResolver object creates a result set with None in it in the first place. Seems like the existing None result was not introduced by cross-training. We will need to trace @taicchoumsft's repro.